cmd/src/dump.rs

```diff
@@ -0,0 +1,108 @@
+use engine_rocks::{
+    self,
+    raw::{DBOptions, Env},
+    RocksEngine,
+};
+use engine_traits::{Iterable, RaftEngine, RaftLogBatch, CF_DEFAULT};
+use kvproto::raft_serverpb::RaftLocalState;
+use protobuf::Message;
+use raft::eraftpb::Entry;
+use std::fs;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+/// Check the potential original raftdb directory and try to dump data out.
+///
+/// Procedure:
+///     1. Check whether the dump has been completed. If there is a flag file,
```

Comment:
I'm confused by the use of flag file. IMO we can rename the raftdb to move it into a trash dir (which should be atomic) and then run `fs::remove_dir_all`. On restarts we can check the trash dir and delete whatever in it.